pkg/process/checks/process.go

```diff
@@ -137,6 +139,10 @@ func (p *ProcessCheck) Init(syscfg *SysProbeConfig, info *HostInfo, oneShot bool
 	}
 	p.containerProvider = sharedContainerProvider
 
+	log.Info("Initializing gpu detector from process check")
+	p.gpuDetector = NewGPUDetector(p.wmeta)
+	log.Info("Finish initializing gpu detector from process check")
```

Comment:
We shouldn't be initializing anything here. We should be passing it in from the constructor and it should be a component. The component will know if it's running as part of the core agent and we can create a real check or just have a no-op if it's running in the process-agent.

Reply:
Will address in #34137